### PR TITLE
added hw 10.1 solution

### PR DESCRIPTION
**Pull request by programmer28**

### CODE REVIEW

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 1]
```diff
@@ -17,7 +27,24 @@
...
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+        if (!new File(fileName).exists()) {
+            throw new FileNotFoundException("The file doesn`t exist");
+        }
+        String string = Files.readString(Paths.get(fileName));
+        string = string.toLowerCase().replaceAll("[^a-z]", "");
+        Set<Character> set = new TreeSet<>();
+        for (char ch : string.toCharArray()) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]: `ch` - what does it mean?

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 2]
```diff
@@ -17,7 +27,24 @@
...
+            set.add(ch);
+        }
+        StringBuilder result = new StringBuilder();
+        List<Character> list = new ArrayList<>(set);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]: should we use ArrayList here? try implementing this method without ArrayList

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 3]
```diff
@@ -17,7 +27,24 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]: let's not use throws IOException

File: src/main/java/core/basesyntax/TreeSetCharacters.java (hunk 1 above)
Comment (programmer28) [reply src/main/java/core/basesyntax/TreeSetCharacters.java]: it means one symbol which is one element from our CharArray. Okay I`ve changed it for symbol

File: src/main/java/core/basesyntax/TreeSetCharacters.java (hunk 2 above)
Comment (programmer28) [reply src/main/java/core/basesyntax/TreeSetCharacters.java]: You are right. I`ve changed this part of code

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 4]
```diff
@@ -17,7 +25,25 @@
...
+        Set<Character> set = new TreeSet<>();
+        for (char symbol : string.toCharArray()) {
+            set.add(symbol);
+        }
+        StringBuilder result = new StringBuilder();
+        int count = 0;
+        for (Character symbol : set) {
+            result.append(symbol);
+            count++;
+            if (count == 5) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]:
```suggestion
            if (result.length >= 5) {
                break;
            }
```

Comment (programmer28) [reply src/main/java/core/basesyntax/TreeSetCharacters.java]: I`ve done it

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**